opal/corelib/runtime.js

```diff
@@ -124,21 +124,139 @@
   //     Opal.Object.$$scope.get("Array")
   //
   // If a constant with the given name cannot be found, then a dispatch to the
-  // class/module's `#const_method` is called, which by default will raise an
+  // class/module's `#const_missing` is called, which by default will raise an
   // error.
   //
   // @param name [String] the name of the constant to lookup
   // @return [Object]
   //
-  Opal.get = function(name) {
-    var constant = this[name];
+  Opal.get = function(const_name) {
+    return Opal.const_get([this], const_name, true, true);
+  }
+
+  // Finds and returns a constant in the provided list of scopes.
+  // When you open a class/module/singleton class, Opal collects
+  // scopes in the $scopes array. When you try to resolve
+  // a constant Opal.const_get is used to get it.
+  //
+  // To simply find a constant directly in the single scope:
+  //
+  //     Opal.const_get([scope], const_name)
+  //
+  // To search in parents:
+  //
+  //     Opal.const_get(scopes, const_name, true)
+  //
+  // To throw an error if nothing was found:
+  //
+  //     Opal.const_get(scopes, const_name, inherit, true)
+  //
+  // @param scopes [Array<$$scope>] a list of scopes
+  // @param const_name [String] the name of the constant
+  // @param inherit [Boolean] flag to perform a search in the parents
+  // @param raise [Boolean] flag to trigger "const_missing" if nothing was found
+  // @return [Object]
+  Opal.const_get = function(scopes, const_name, inherit, raise) {
+    function getSingle(scope, const_name, inherit) {
```

Comment:
@iliabylich Can any of these functions be moved outside to avoid multiple evaluation?

(Please correct me if you already know that it doesn't affect perfs!)